lib/skate/detours/shape_segment.ex

```diff
@@ -0,0 +1,13 @@
+defmodule Skate.Detours.ShapeSegment do
+  @moduledoc """
+  Represents the shape points which form a segment associated
+  with stops on the shape.
+  """
+
+  @type t :: %__MODULE__{
+          points: [Util.Location.From.t()],
+          stops: [Util.Location.From.t()]
```

Comment:
If I remember correctly, `ShapeSegment` will never have more than one stop, right? If so, could we make that more explicit in the type here? Something like:

```suggestion
          stops: {:some, Util.Location.From.t()} | {:none}
```

Or even

```suggestion
          stop: Util.Location.From.t() | nil
```

Reply:
That would necessarily mean changing the `stops` invocations to use singular stops instead of arrays, but I think that would be an improvement, since there's never more than one stop.

Reply:
> If I remember correctly, `ShapeSegment` will never have more than one stop, right? If so, could we make that more explicit in the type here?

_currently_ `ShapeSegment` will not have more than one stop, but, this is something that I 100% believe could happen as we improve the algorithm. 

I personally would keep it as is, but I'm maybe trying to over-fit for the (unknown) future.

Reply:
@joshlarson thoughts? 2b5f726494f0dbc21a1cccd105059734daba6cf5